### PR TITLE
Add Atlassian case study

### PR DESCRIPTION
**Pull request by zchase**
This PR adds the Atlassian case study.

### CODE REVIEW

File: themes/default/content/case-studies/atlassian.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    and reduced developers' time spent on maintenance by 50%.
+
+customer_name: Atlassian
+customer_logo: /logos/customers/atlassian-wordmark.svg
+customer_url: https://www.atlassian.com/
+
+exec_summary: |
+    Atlassian’s Bitbucket DevSpeed team is responsible for improving developer productivity through better workflows
+    and tooling. Because the team wanted to make it easier and faster for developers to access cloud infrastructure,
+    they moved from a legacy DSL-based infrastructure as code tool  to the Pulumi Cloud Engineering Platform, which
```
Comment (jaxxstorm) [inline themes/default/content/case-studies/atlassian.md:17]:
```suggestion
    they moved from a legacy DSL-based infrastructure as code tool to the Pulumi Cloud Engineering Platform, which
```

**Review by jaxxstorm — APPROVED** [review]

File: themes/default/content/case-studies/atlassian.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Once the instance is ready, the developer simply SSH’s into it, and it will look exactly like Bitbucket. The developer can make the changes they want and then create a pull request to merge their work into a main branch. Developers can also share the URL of the instance with teammates to see the changes when they review the code.
+
+Here is what an instance, spun up with Pulumi, looks like:
+
+<img class="block mx-auto md:max-w-4xl my-8"
+src="/images/case-studies/bitbucket-cloud-diagram.png" alt="Bitbucket Cloud and Pulumi">
+
+### Using the Automation API
+
+The [Pulumi Automation API]({{< relref "/docs/guides/automation-api" >}}) exposes the full power of infrastructure as code through a programmatic interface instead of CLI commands. It exposes Pulumi programs and stacks as strongly-typed and composable building blocks, enabling developers to automate deployments directly from their code at run-time. (A Pulumi program, which is written in your chosen programming language, describes how your cloud infrastructure should be composed.)
```
Comment (gthuang) [inline themes/default/content/case-studies/atlassian.md:99]: this link is not working in the preview either

File: themes/default/content/case-studies/atlassian.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+### Results
+
+Using Pulumi with its existing CI/CD process, the DevSpeed team built a self-service dashboard that enables any Bitbucket developer to quickly and easily provision a cloud-based development environment. Compared to the DSL-based tool they previously used, Pulumi significantly reduced the number of lines of code needed and made it much easier to read and understand. In terms of user satisfaction, the new dashboard meant that developers went from spending one day a week maintaining their environment to less than four hours per week. Mike’s team also receives significantly fewer infrastructure questions from developers on their Slack help channel.
```
Comment (gthuang) [inline themes/default/content/case-studies/atlassian.md:105]:
```suggestion
Using Pulumi with its existing CI/CD process, the DevSpeed team built a self-service dashboard that enables any Bitbucket developer to quickly and easily provision a cloud-based development environment. Compared to the DSL-based tool they previously used, Pulumi significantly reduced the number of lines of code needed to model infrastructure and made the code much easier to read and understand. In terms of user satisfaction, the new dashboard meant that developers went from spending one day a week maintaining their environment to less than four hours per week. Mike’s team also receives significantly fewer infrastructure questions from developers on their Slack help channel.
```

File: themes/default/content/case-studies/atlassian.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+## Summary
+
+Here is a summary of how Pulumi helped Bitbucket and the DevSpeed team.
+
+- Allowed developers to use general purpose programming languages like Python and familiar software tools to deliver and manage infrastructure
```
Comment (gthuang) [inline themes/default/content/case-studies/atlassian.md:111]:
```suggestion
- Allowed developers to use general purpose programming languages like Python and familiar software tools to build and manage infrastructure as code
```

File: themes/default/content/case-studies/atlassian.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+## Second Challenge: Improving Developer Productivity with Self-Service Cloud Infrastructure
+
+Shortly afterward, Mike reviewed a survey sent to the Bitbucket developers, asking them how they spent their time. He was surprised to learn that developers were spending 20% of their workweek maintaining their private Bitbucket development environments.
+
+Again, part of the problem rested with the old tool, whose DSL was unfamiliar to most developers. Because it lacked standard programming capabilities, it made the code verbose and difficult to understand. Mike knew that he could streamline the entire workflow with Pulumi and make infrastructure more accessible and simpler to use for developers who understand Python.
+
+To make things simple and turnkey for developers, the DevSpeed team built a self-service dashboard that created and configured new instances. Developers enter a few pieces of information, such as what they want to call the instance, what region to use, and an SSH key. Then, a CI/CD pipeline using Bitbucket Pipelines and Bamboo calls Pulumi to deploy a complete developer environment.
```
Comment (gthuang) [inline themes/default/content/case-studies/atlassian.md:88]:
```suggestion
To make things simple and turnkey for developers, the DevSpeed team built a self-service dashboard that created and configured new Bitbucket instances. Developers enter a few pieces of information, such as what they want to call the instance, what region to use, and an SSH key. Then, a CI/CD pipeline using Bitbucket Pipelines and Bamboo calls Pulumi to deploy a complete developer environment.
```

File: themes/default/content/case-studies/atlassian.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+customer_url: https://www.atlassian.com/
+
+exec_summary: |
+    Atlassian’s Bitbucket DevSpeed team is responsible for improving developer productivity through better workflows
+    and tooling. Because the team wanted to make it easier and faster for developers to access cloud infrastructure,
+    they moved from a legacy DSL-based infrastructure as code tool  to the Pulumi Cloud Engineering Platform, which
+    let them define and deploy infrastructure in general purpose languages that Bitbucket developers already used,
+    such as Python. Using a familiar language like Python also made it easy for them to  add cross-regional support
+    to their CI/CD pipeline for deploying development environments for over 100 Bitbucket developers around the world.
+    They also built a self-service dashboard that allows any developer to deploy and configure instances for feature
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:21]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    maintaining their instances by more than 50%.
+
+
+
+sections:
+    - label: Exec Summary
+      anchor: executive-summary
+    - label: About Bitbucket Cloud
+      anchor: about-bitbucket-cloud
+    - label: "First Challenge: Enabling the Sydney Developers"
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:32]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      anchor: first-challenge-enabling-the-sydney-developers
+    - label: "Second Challenge: Improving Developer Productivity with Self-Service Cloud Infrastructure"
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:34]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+He cited an example, saying, “We have three databases. With the old tool, that was 20 lines of code. And of course, coming from a C# background, I had no idea what any of it meant. It would take me hours to figure out how any of it worked and what each property name was and what it did. Another confusion I had was, defining a variable “name” in multiple places, and I had no idea which one was correct. So many things were unclear.”
+
+Mike wanted a general and elegant solution that would let him quickly and easily manage infrastructure in AWS while using familiar languages and tools.
+
+### Pulumi to the Rescue
+
+Mike says, “I knew about Pulumi because I followed Joe Duffy on Twitter. I come from the C# world, where he's a well-known figure. I saw that he’d started a company called Pulumi and its mission really appealed to me.”
+
+Pulumi allows developers to build, deploy, and manage cloud infrastructure and applications using the programming language of their choice. That means they can use standard programming constructs and standard tools to author and manage infrastructure code.
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:66]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+Mike chose Python to build cloud infrastructure since Bitbucket developers already used it. He said, “Because all the developers on Bitbucket write Python, they can use all the skills they already have from their day-to-day job to write infrastructure code. There are also safety features you’d expect from a standard programming language. For example, you know if you're referencing something that isn't declared or if you're trying to use a property that doesn't exist.”
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:68]: 👍

File: themes/default/content/case-studies/atlassian.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (gthuang) [reply themes/default/content/case-studies/atlassian.md:21]: sorry, actually make that "Bitbucket instances"

File: themes/default/content/case-studies/atlassian.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+As an example of how Pulumi made formerly complex tasks straightforward, Mike says, “With the old tool, spinning up our databases meant we had to have about 20 blocks of code and do a lot of copy and pasting. But with Pulumi, it's Python. So you have a for statement and then you define the names of the databases you want. It's five lines of code. If you want to add a new database, add one line, and you're good to go.”
+
+Pulumi also simplified cloud infrastructure management and provided clear visibility into all active resources. Mike said, “Before Pulumi, we had a manager for state files that was a complete pain. We ran into weird issues and errors, and it was difficult to ask questions, such as how many developer instances were running. Now, Pulumi manages all the state files for us and gives us that visibility.""
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:74]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+Mike also gave an example of how easy it was to deploy changes across all instances with confidence. He says, “Subnets have a maximum number of IP addresses associated with them. We ran into an issue where there would be failures because addresses were selected randomly. I corrected the issue and could confidently deploy it by just writing a test to make sure that select logic worked. And because we're using Python, I was able to just use the AWS Python API to figure out which subnets were full! That's simply not possible using a DSL.”
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:76]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+### Result
+
+Using Pulumi, with its support for general-purpose languages like Python, Mike quickly and easily built a repeatable pipeline that deploys development environments for the Sydney team despite having limited knowledge of AWS. He also now understood how useful Pulumi could be in improving the developer experience at Bitbucket and how easy it was to use. He began to look for other ways to use Pulumi.
+
+## Second Challenge: Improving Developer Productivity with Self-Service Cloud Infrastructure
+
+Shortly afterward, Mike reviewed a survey sent to the Bitbucket developers, asking them how they spent their time. He was surprised to learn that developers were spending 20% of their workweek maintaining their private Bitbucket development environments.
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:84]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+Again, part of the problem rested with the old tool, whose DSL was unfamiliar to most developers. Because it lacked standard programming capabilities, it made the code verbose and difficult to understand. Mike knew that he could streamline the entire workflow with Pulumi and make infrastructure more accessible and simpler to use for developers who understand Python.
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:86]: 👍

File: themes/default/content/case-studies/atlassian.md (hunk 5 above)
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:88]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+To make things simple and turnkey for developers, the DevSpeed team built a self-service dashboard that created and configured new instances. Developers enter a few pieces of information, such as what they want to call the instance, what region to use, and an SSH key. Then, a CI/CD pipeline using Bitbucket Pipelines and Bamboo calls Pulumi to deploy a complete developer environment.
+
+Once the instance is ready, the developer simply SSH’s into it, and it will look exactly like Bitbucket. The developer can make the changes they want and then create a pull request to merge their work into a main branch. Developers can also share the URL of the instance with teammates to see the changes when they review the code.
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:90]: 👍

File: themes/default/content/case-studies/atlassian.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+Here is what an instance, spun up with Pulumi, looks like:
```
[the remark this answers is not in the document]
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:92]: 👍

File: themes/default/content/case-studies/atlassian.md (hunk 3 above)
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:105]: 👍

File: themes/default/content/case-studies/atlassian.md (hunk 4 above)
Comment (zchase) [reply themes/default/content/case-studies/atlassian.md:111]: 👍

**Pull request merged**